### DISCUSSION TITLE
`setup quota set ...` does not work if account has already exceeded quota?

### DISCUSSION THREAD

**Discussion by leoborlot** [docker-mailserver/docker-mailserver#3633] [Q&A]
Hi guys. 

I have a problem with 2 overquote email accounts

I try set a new Quota using the setup. Everthings look ok, the dovecot-quotas.cf whans change, but, when I call 'email list' the quota dont was change. 

When I open with roundcube, the quota was´snt change too... Where was the problem?

When I try with anothe accounts, no overquote this work...

**Comment by leoborlot** [comment]
No way, I create a new account and this accept to change quota, but, my old email account, dont

**Comment by leoborlot** [comment]
another information. I del the email account, restart the server, but, when I try to create again, this start with 15GB...

**Comment by polarathene** [comment]
- [`setup quota set ...` command](https://github.com/docker-mailserver/docker-mailserver/blob/26214491efb2ec07242c24a9191633f0d72998d9/target/bin/setquota#L14) should [update the file](https://github.com/docker-mailserver/docker-mailserver/blob/26214491efb2ec07242c24a9191633f0d72998d9/target/scripts/helpers/database/manage/dovecot-quotas.sh#L15) (`dovecot-quotas.cf`) on disk correctly.
- That [file is monitored](https://github.com/docker-mailserver/docker-mailserver/blob/26214491efb2ec07242c24a9191633f0d72998d9/target/scripts/helpers/change-detection.sh#L40) by our `check-for-changes.sh` service script [which should process it when updated](https://github.com/docker-mailserver/docker-mailserver/blob/26214491efb2ec07242c24a9191633f0d72998d9/target/scripts/check-for-changes.sh#L106).
- Updates may take a bit of time, as this has a very naive implementation that creates every single account from config files again and reloads Postfix + Dovecot. If you use a remote filesystem like NFS this may be less reliable too.
- Ensure that the file change has also been visible within the container `docker exec -it container_name_here cat /tmp/docker-mailserver/dovecot-quotas.cf`. If it hasn't there is a problem with your volume config. This should be unlikely as the file change is applied in the container directly.
- Both Dovecot quota and the change detection service features are not available for LDAP users presently.

---

You should provide more information about your setup so that it can be reproduced.

You mention RoundCube, if this is changing quota from UI it is not something supported by DMS as I doubt it would be changing the `dovecot-quotas.cf` file and DMS would not be aware of any other software changing Dovecot accounts so all those third-party changes would be lost when DMS updates Dovecot accounts.

---

The main issue you are describing is that an account has exceeded it's quota, you used the `setup ...` command to increase the quota and `dovecot-quotas.cf` is updated, but `setup email list` is not showing the updated quota? How long did you wait? Should be done within 5-10 seconds depending on your server / storage driver.

If you can always reproduce this problem with account that has exceeded quota, and never with account that has enough quota remaining, then it is likely a bug and will need a bug report with reproduction steps.

**Reply by leoborlot** [reply]
Lets me try, there some thing wrong persisting the data, i will come back with awsher
